### ISSUE TITLE
Have you try to add linguistic features in your nmt experiment? I need some help.

### ISSUE THREAD

**Issue by hfxunlp** [harvardnlp/seq2seq-attn#85]
I try to add pos-tagging with chinese-english machine translation, but get a 4+ BLEU score losses(baseline:34+, add pos-tagging:29+). And I do not know why, I need to put some features into nmt system, but I do not know how to do it now. Thank you and hope you could help me.
This is my training command:
th train.lua -data_file data/tseg-train.hdf5 -val_data_file data/tseg-val.hdf5 -savefile tseg-model -num_layers 1 -rnn_size 1000 -word_vec_size 620 -reverse_src 1 -dropout 0.3 -lr_decay 0.5 -attn 1 -optim sgd -learning_rate 1 -start_decay_at 1 -epochs 50 -max_batch_l 80 -gpuid 1 -feature_embeddings_dim_exponent 0.7 | tee tblog.txt

**Comment by jsenellart** [comment]
Hi Ano! Can you share a sample of your training data where you added pos-tagging so that we can advice you. Also - could you try using OpenNMT project which is the new version of `seq2seq-attn` - and you can also join the forum (forum.opennmt.net) to ask some help to the community.

**Comment by hfxunlp** [comment]
Thank you for your advise, I'm trying OpenNMT now. Here are some lines of my training data and I do not know whether it was enough:

载入-|-v 黛-|-ng 妃-|-ng 死因-|-n 调查-|-vn 资料-|-n 的-|-ud 两-|-m 台-|-q 手提-|-b 电脑-|-n 遭-|-v 窃-|-vg
法新社-|-nt 伦敦-|-ns 六日-|-t 电-|-n
伦敦-|-ns 每日-|-r 快报-|-n 指出-|-v ,-|-wd 两-|-m 台-|-q 记载-|-n 黛-|-ng 安娜-|-nr 王妃-|-n 一九九七年-|-t 巴黎-|-ns 死亡-|-vi 车祸-|-n 调查-|-vn 资料-|-n 的-|-ud 手-|-n 提-|-v 电脑-|-n ,-|-wd 被-|-p 从前-|-t 大都会-|-n 警察-|-n 总长-|-n 的-|-ud 办公室-|-n 里-|-f 偷走-|-v 。-|-wj
报导-|-v 指出-|-v ,-|-wd 这-|-rzw 起-|-q 窃案-|-n 引发-|-v 人们-|-n 猜测-|-vn ,-|-wd 电脑-|-n 中-|-f 可能-|-v 藏-|-v 有-|-v 由-|-p 前-|-f 大都会-|-n 警察-|-n 总长-|-n 史蒂文森-|-nr 爵士-|-n 领衔-|-vd 调查-|-v 的-|-ud 机密-|-n 资料-|-n 。-|-wj
不过-|-c ,-|-wd 一-|-m 名-|-q 大都会-|-n 警察局-|-n 女-|-b 发言人-|-n 表示-|-v ,-|-wd 电脑-|-n 里-|-f 并-|-d 没有-|-v 任何-|-rz 敏感-|-a 的-|-ud 资讯-|-n ,-|-wd 或-|-c 与-|-p 调查-|-v 相关-|-vi 的-|-ud 资料-|-n 。-|-wj

Two Laptops with Inquiry Data on the Cause of Diana's Death Stolen
AFP on the 6th, London
London's Daily Express noted that two laptops with inquiry data on the 1997 Paris car accident that caused the death of Princess Diana were stolen from the office of a former metropolitan police commissioner.
The report pointed out that the theft triggered speculation that the computers may contain confidential information of the probe led by former Metropolitan Police Commissioner Lord Stevens.
However, a Metropolitan Police spokeswoman said that there was no sensitive information or any data linked to the probe in the computers.

It is Chinese-English translation and there are 70+ pos-tagging in Chinese.
Sorry for reply so late, my workstation at school just get power yestoday so I can get the data out. @jsenellart-systran

**Comment by hfxunlp** [comment]
In fact some of my teachers and me want to add Chinese function words' usages to NMT and hope it could help the tranlation. It is very special in Chinese that function words controls syntactical structure while content words express the meaning. We have data and tools but problems occurs while we are trying to add it to NMT.

**Comment by hfxunlp** [comment]
Maybe my data was bigger than your example? I have 1.25 million sentence pairs. The modules just got trained enough good with 5~7 epochs.

**Comment by srush** [comment]
Can you move this conversation to the forum for OpenNMT at forum.opennmt.net?
Thanks!

**Comment by hfxunlp** [comment]
ok， I will. @srush
